src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method uses `indexOf` and `splice` which can be inefficient when dealing with large arrays. Instead, consider using the `filter` method to remove dead animals from the `Animal.alive` array. This approach avoids nesting loops and can improve performance.